padl/utils.py

```diff
@@ -61,3 +63,131 @@ def __getattr__(self, attr):
 
 #: Transform factory for capturing attributes/ get-items.
 same = _Same()
+
+
+class _Debug:
+    """Customized debugger for :class:`padl.transforms.Transform`s.
+
+    When an exception on the execution of a :class:`padl.transforms.Transform` is produced and a
+    :class:`_Debug` object is called, an interactive debugger at different levels in the
+    :class:`padl.transforms.Transform` is gotten.
+
+    At the top, the user interacts with the entire transform and its absolute input. One level
+    down, it goes directly to the stage that got the Exception (either to
+    :meth:`padl.transforms.Transform.pd_preprocess`, :meth:`padl.transforms.Transform.pd_forward`,
+    or :meth:`padl.transforms.Transform.pd_postprocess`) and each level deeper moves recursively
+    inside the element that failed until the :class:`padl.transforms.AtomicTransform` that got the
+    Exception is reached.
+    """
+    def __init__(self):
+        self.trans = None
+        self.args = None
+        self.default_msg = (
+            'Defined commands are: \n'
+            '   u(p): step up\n'
+            '   d(own): step down\n'
+            '   w(here am I?): show code position\n'
+            '   i(nput): show input here\n'
+            '   r(epeat): repeat here (will produce the same exception)\n'
+            '   t(ransform): displays the current transform\n'
+            '   h(elp): print help about the commands\n'
+            '   q(uit): quit'
+            '     -> this will store the input at this level in debug.args\n'
+            '        and the transform in debug.trans\n'
+        )
+
+    def __call__(self) -> None:
+        """Call me for getting an interactive debugger in case of error.
+
+        User can give following input and expect response
+            u(p): step up\n'
```

Comment:
I don't think you need the newline characters here since you have used a triple quote to wrap the doc string.